### PR TITLE
crt clustering clusterreco

### PR DESCRIPTION
**Pull request by henrylay97**
Following extensive work to rewrite the CRT reconstruction for SBND the PR strategy has been agreed to be a series of PRs against a common branch (`feature/hlay_crt_clustering_base`) for the (inevitably very long) review stage. Please don't use CI tests yet. They will be used on the final PR when the common branch is compared to `develop`. At this stage the individual branches may not each compile on their own. The fully merged branch can be viewed here for completeness [feature/hlay_crt_clustering_merged](https://github.com/SBNSoftware/sbndcode/tree/feature/hlay_crt_clustering_merged).

This PR introduces the clustering stage of the reconstruction, grouping CRTStripHits in a single tagger into single objects if their positioning and timing suggests they should be. The relevant object is created in SBNSoftware/sbnobj#83.

### CODE REVIEW

File: sbndcode/CRT/CRTReco/CRTClusterProducer_module.cc [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+          clusteredHits.push_back(initialStripHit);
+          used[i] = true;
+
+          for(uint16_t ii = i+1; ii < stripHits.size(); ++ii)
+            {
+              const art::Ptr<CRTStripHit> &stripHit = stripHits[ii];
+    
+              if(!used[ii])
+                {
+                  if(stripHit->Ts1() - initialStripHit->Ts1() < fCoincidenceTimeRequirement)
```
Comment (absolution1) [inline sbndcode/CRT/CRTReco/CRTClusterProducer_module.cc:147]: Is stripHit Ts1 guaranteed to be greater than initialStripHit Ts1?

**Review by absolution1 — APPROVED** [review]
Looks good.
I'm approving this, but did have one question about the logic.

File: sbndcode/CRT/CRTReco/CRTClusterProducer_module.cc (hunk 1 above)
Comment (henrylay97) [reply sbndcode/CRT/CRTReco/CRTClusterProducer_module.cc:147]: Great catch but yeah the strip hit vector is sorted by T1 value before this function is called :D

Comment (henrylay97) [reply sbndcode/CRT/CRTReco/CRTClusterProducer_module.cc:147]: (Line 89 if anyone's looking)

**Pull request merged**